### ISSUE TITLE
Allow UUID-style digit grouping

### ISSUE THREAD

**Issue by Diggsey** [rust-lang/rust-clippy#5431]
The lint "inconsistent-digit-grouping" should allow things that look like UUIDs, eg:

```rust
0xc43bbeba_12b6_4868_80bc_4bd324dbe4e1
```

And possibly other "common" number patterns.
